src/ranged.cpp

```diff
@@ -509,7 +504,7 @@ void player::fire_gun( const tripoint &targ_arg, bool burst )
         } else if ( used_weapon->deactivate_charger_gun() ) {
             // Deactivated charger gun
         } else {
-            used_weapon->charges -= used_weapon->ammo_required();
+            used_weapon->ammo_consume( used_weapon->ammo_required() );
```

Comment:
This is the only place where you use `ammo_consume` and you drop the value it returns.
Shouldn't a debugmsg be printed on failure or something?
What about weapons that consume 0 ammo? Those will result in `ammo_consume` returning `false`.